app/src/main/java/com/github/multimatum_team/multimatum/procrastination_detector/ProcrastinationDetectorService.kt

```diff
@@ -0,0 +1,82 @@
+package com.github.multimatum_team.multimatum.procrastination_detector
+
+import android.app.Service
+import android.content.Intent
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.os.IBinder
+import android.widget.Toast
+import com.github.multimatum_team.multimatum.R
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import kotlin.math.abs
+
+@AndroidEntryPoint
+class ProcrastinationDetectorService : Service(), SensorEventListener {
+    @Inject lateinit var sensorManager: SensorManager
+
+    // TODO make it remain enabled when app is stopped
+
+    private var lastDetection: Long = 0L
+    private var lastPosition: Array<Float>? = null  // null only at initialization
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        val refSensor = sensorManager.getDefaultSensor(REF_SENSOR)
+            ?: throw IllegalStateException("missing sensor")
+        super.onCreate()
+        sensorManager.registerListener(this, refSensor, SensorManager.SENSOR_DELAY_NORMAL)
+        toast("Procrastination fighter is enabled")
+        return START_STICKY  // service must restart as soon as possible if preempted
+    }
+
+    override fun onBind(intent: Intent): IBinder {
+        throw UnsupportedOperationException("cannot bind procrastinationDetectorService: not implemented")
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        val refSensor = sensorManager.getDefaultSensor(REF_SENSOR)
+        sensorManager.unregisterListener(this, refSensor)
+        toast("Procrastination fighter is disabled")
+    }
+
+    override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) { /* Nothing to do */ }
+
+    override fun onSensorChanged(event: SensorEvent?) {
+        val currentTime = System.currentTimeMillis()
+        if (currentTime >= lastDetection + MIN_PERIOD_BETWEEN_NOTIF){
+            val currentPosition = event!!.values.toTypedArray()
+            if (lastPosition != null && l1Distance(currentPosition, lastPosition!!) > MOVE_DETECTION_THRESHOLD) {
```

Comment:
As I see, a Toast appear when you move from where you where before. However, a distance of 0.1 is really short. You could do a movment like this when you add a deadline in Mulmatum, which will render the feature annoying even in the app.

Reply:
0.1 is a threshold that I determined experimentally. It has no unit because it refers to the values given by the sensor, which don't have units. And yes, currently it displays the toast even if the movement comes from setting a deadline in the app. Eventually the goal will probably be to avoid this behavior, e.g. by stopping the service when the app is used. But the current implementation is incomplete and a bit experimental, and does not support this complex behavior. But you are right about the fact that this is confusing, I'll add a comment.

Reply:
Okay, thanks for the precisions :)